lib/ruby_indexer/lib/ruby_indexer/entry.rb

```diff
@@ -40,6 +40,21 @@ class Namespace < Entry
 
       abstract!
 
+      sig { returns(Entry::Class) }
+      attr_reader :singleton_klass
+
+      sig { params(name: String, file_path: String, location: Prism::Location, comments: T::Array[String]).void }
+      def initialize(name, file_path, location, comments)
+        super
+
+        singleton = if name.start_with?("Class:")
+          T.cast(self, Entry::Class)
```

Comment:
Why do we need this recursive link?

Reply:
Because the singleton class of a singleton class is itself. And if we didn't have the check we would have an infinite loop.

Reply:
Yeah I think creating the singleton class only when we need it makes more sense